From 0c2dfb9238ce0978079f7a430ef81ba3540699bd Mon Sep 17 00:00:00 2001
From: Kyle McMartin <kmcmarti@redhat.com>
Date: Fri, 12 Sep 2014 17:46:40 -0400
Subject: [kernel] modules: export check_module_rhelversion

Message-id: <20140912174640.GU29420@redacted.bos.redhat.com>
Patchwork-id: 94056
O-Subject: [RHEL7.1 PATCH 4/6 v4] modules: export check_module_rhelversion
Bugzilla: 1108378
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Johannes Weiner <jweiner@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

Export check_module_rhelversion so we can use it outside of module.c,
needed by x86_64's module loader to determine if we're loading a RHEL
7.0 module which needs a KABI fix up for 16K stacks on RHEL 7.1

Signed-off-by: Kyle McMartin <kmcmarti@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/module.h b/include/linux/module.h
index 504035f..9880f37 100644
--- a/include/linux/module.h
+++ b/include/linux/module.h
@@ -514,6 +514,8 @@ int unregister_module_notifier(struct notifier_block * nb);
 
 extern void print_modules(void);
 
+bool check_module_rhelversion(struct module *mod, char *version);
+
 #else /* !CONFIG_MODULES... */
 
 /* Given an address, look for it in the exception tables. */
@@ -624,6 +626,12 @@ static inline int unregister_module_notifier(struct notifier_block * nb)
 static inline void print_modules(void)
 {
 }
+
+static inline bool check_module_rhelversion(struct module *mod, char *version)
+{
+ return false;
+}
+
 #endif /* CONFIG_MODULES */
 
 #ifdef CONFIG_SYSFS
-- 
1.7.1